### PR TITLE
Add latest SZ filter to list of plugins

### PR DESCRIPTION
**Pull request by byrnHDF**
Special recoded SZ filter tar.gz file included, which adds options to CMake files and adds configure defines for headers.

### CODE REVIEW

File: BSHUF/config/cmake/CTestScript.cmake [hunk 1]
```diff
@@ -25,30 +25,23 @@ if (NOT SITE_OS_NAME)
...
+  if (SITE_BUILDNAME_SUFFIX)
+    set (CTEST_BUILD_NAME  "${SITE_BUILDNAME_SUFFIX}-${CTEST_BUILD_NAME}")
+  endif ()
+  set (BUILD_OPTIONS "${ADD_BUILD_OPTIONS}")
 else ()
   ## machine name provided
   ## --------------------------
   if (CMAKE_HOST_UNIX)
-    set (CTEST_BUILD_NAME "${SITE_OS_NAME}-${SITE_OS_VERSION}-${SITE_OS_BITS}-${SITE_COMPILER_NAME}-${SITE_COMPILER_VERSION}")
+    set(CTEST_BUILD_NAME "${SITE_OS_NAME}-${SITE_OS_VERSION}-${SITE_OS_BITS}-${SITE_COMPILER_NAME}-${SITE_COMPILER_VERSION}")
```
Comment (lrknox) [inline BSHUF/config/cmake/CTestScript.cmake:36]: Why is the space between set and "(CTEST..." removed from lines 32 & 34 while new "set (CTEST..." lines in lines 31 & 43 have the space between?

File: BSHUF/config/cmake/ConfigureChecks.cmake [hunk 2]
```diff
@@ -17,7 +17,7 @@ if (APPLE)
   list (LENGTH CMAKE_OSX_ARCHITECTURES ARCH_LENGTH)
   if (ARCH_LENGTH GREATER 1)
     set (CMAKE_OSX_ARCHITECTURES "" CACHE STRING "" FORCE)
-    message(FATAL_ERROR "Building Universal Binaries on OS X is NOT supported by the H5BSHUF project. This is"
+    message (FATAL_ERROR "Building Universal Binaries on OS X is NOT supported by the H5BZ2 project. This is"
```
Comment (lrknox) [inline BSHUF/config/cmake/ConfigureChecks.cmake:20]: Is this change intended in a BSHUF subdirectory?

Comment (byrnHDF) [reply BSHUF/config/cmake/ConfigureChecks.cmake:20]: no

File: JPEG/config/cmake/ConfigureChecks.cmake [hunk 3]
```diff
@@ -17,7 +17,7 @@ if (APPLE)
   list (LENGTH CMAKE_OSX_ARCHITECTURES ARCH_LENGTH)
   if (ARCH_LENGTH GREATER 1)
     set (CMAKE_OSX_ARCHITECTURES "" CACHE STRING "" FORCE)
-    message(FATAL_ERROR "Building Universal Binaries on OS X is NOT supported by the H5JPEG project. This is"
+    message (FATAL_ERROR "Building Universal Binaries on OS X is NOT supported by the H5BZ2 project. This is"
```
Comment (lrknox) [inline JPEG/config/cmake/ConfigureChecks.cmake:20]: Another change to H5BZ2 in a subdirectory of a different filter (JPEG).

File: BSHUF/config/cmake/ConfigureChecks.cmake (hunk 2 above)
Comment (byrnHDF) [reply BSHUF/config/cmake/ConfigureChecks.cmake]: fixed

File: BSHUF/config/cmake/CTestScript.cmake (hunk 1 above)
Comment (byrnHDF) [reply BSHUF/config/cmake/CTestScript.cmake:36]: no reason

**Review by lrknox — APPROVED** [review]
